pandas/io/common.py

```diff
@@ -532,16 +535,61 @@ def check_parent_directory(path: Path | str) -> None:
         raise OSError(fr"Cannot save file into a non-existent directory: '{parent}'")
 
 
+@overload
 def get_handle(
     path_or_buf: FilePathOrBuffer,
     mode: str,
+    *,
+    encoding: str | None = ...,
+    compression: CompressionOptions = ...,
+    is_text: Literal[True],
+    memory_map: bool = ...,
+    errors: str | None = ...,
+    storage_options: StorageOptions = ...,
+) -> IOHandles[str]:
+    ...
+
+
+@overload
+def get_handle(
+    path_or_buf: FilePathOrBuffer,
+    mode: str,
+    *,
+    encoding: str | None = ...,
+    compression: CompressionOptions = ...,
+    is_text: Literal[False],
+    errors: str | None = ...,
+    memory_map: bool = ...,
+    storage_options: StorageOptions = ...,
+) -> IOHandles[bytes]:
+    ...
+
+
+@overload
+def get_handle(
+    path_or_buf: FilePathOrBuffer,
+    mode: str,
+    *,
+    encoding: str | None = ...,
+    compression: CompressionOptions = ...,
+    memory_map: bool = ...,
```

Comment:
doesn't this need is_text: Literal[False] ?

Reply:
I fixed the overloads. I used three overloads but needed only two overloads.